### DISCUSSION TITLE
Best practices to use secrets in spacy projects

### DISCUSSION THREAD

**Discussion by krlng** [explosion/spaCy#7950] [Help: Best practices]
Hi :)

spacy projects will log all commands executed. However, some of those commands might contain sensible information such as passwords. Is there a way to exclude either those variables from logging or deactivate the logging for some commands?

**Comment by polm** [comment]
There's no feature to hide variables or turn off logging for individual commands. 

In this case I would recommend you pass the secrets through some method besides the command line, such as storing them in a file or environment variables set before calling spaCy projects. This is similar to how Github Actions works and [their guidance](https://docs.github.com/en/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow) on using secrets in workflows may be helpful.
